src/java.base/share/classes/java/util/Arrays.java

```diff
@@ -7383,7 +7383,7 @@ public static int compare(double[] a, int aFromIndex, int aToIndex,
      *         lexicographically greater than the second array
      * @since 9
      */
-    public static <T extends Comparable<? super T>> int compare(T[] a, T[] b) {
+    public static <T extends Comparable<? super T>> int compare(T @Nullable [] a, T @Nullable [] b) {
```

Comment:
The Javadoc for this method talks about null array elements. Should the upper bound for `T` have `@Nullable`?

Reply:
Oh, thank you for catching that. I agree that we want `@Nullable` here, and I did not read nearly carefully enough.

I've edited to change the parameter types to `@Nullable T @Nullable []`. I think that works better than `<T extends @Nullable Comparable<? super T>>` because `? super T` would be considered out of bounds if we ran this code through the checker (not that we do that):

```
Compare.java:4: error: [bound.type.incompatible] incompatible bounds in wildcard
  static <T extends @Nullable Comparable<? super T>> int compare(T @Nullable [] a, T @Nullable [] b) {
                                         ^
  type: ? extends @NonNull Object super T extends @Nullable Comparable<?>
  upper bound: @NonNull Object
  lower bound: T[ extends @Nullable Comparable<?[ extends @NonNull Object super T]> super @NonNull Void]
```